rad/prelude/util.rad

```diff
@@ -0,0 +1,8 @@
+(def make-counter
+  "Creates a stateful counter. Returns a dict with two keys: the function at `:next-will-be`
+  will return the next number (without incrementing it), while the function at `:next`
+  increments the number and returns it."
+  (fn []
+    (def i (ref -1))
+    {:next-will-be (fn [] (+ 1 (read-ref i)))
```

Comment:
I think we generally should try to make things more private - here in theory someone could accidentally modify `next-will-be` or `next` and get things wrong. I'm imagining:

```
(def make-counter
   (fn []
     (def i (ref -1))
     (fn [command]
       (if (eq? command :next) (modify-ref ...)
           (if eq? command :current) ....)))
```

And then:

```
(def counter (make-counter))
(counter :current) ;; ==> 0
(counter :next) 
(counter :current) ;; ==> 1
```

What do you think?

Reply:
> someone could accidentally modify `next-will-be`

Not sure what you mean? They are immutable functions how could they be modified? In any case I think what you propose is a little clearer so will change.